src/test/java/com/onlinebookstore/service/BookServiceTest.java

```diff
@@ -57,4 +68,74 @@ public void findById_validBookId_returnBookDto() {
         verify(bookRepository, times(1)).findById(bookId);
         verify(bookMapper, times(1)).toDto(book);
     }
+
+    @Test
+    public void findAll_validBookDto_returnBookDto() {
+        Book book1 = new Book();
+        Book book2 = new Book();
+        List<Book> books = List.of(book1, book2);
+        BookDto bookDto1 = new BookDto();
+        BookDto bookDto2 = new BookDto();
+        Pageable pageable = PageRequest.of(0, 10);
+        Page<Book> bookPage = new PageImpl<>(books, pageable, books.size());
+        when(bookRepository.findAll(pageable)).thenReturn(bookPage);
+        when(bookMapper.toDto(book1)).thenReturn(bookDto1);
+        when(bookMapper.toDto(book2)).thenReturn(bookDto2);
+        List<BookDto> expectedBooks = List.of(bookDto1, bookDto2);
+        List<BookDto> result = bookService.findAll(pageable);
+        assertEquals(expectedBooks, result);
+        verify(bookRepository, times(1)).findAll(pageable);
+        verify(bookMapper, times(1)).toDto(book1);
+        verify(bookMapper, times(1)).toDto(book2);
```

Comment:
Let's initialize book1 and book2 with some values to somehow identify them if test goes wrong. Also use constraction Given When Then with additional space between them to make your test more readable.
See everywhere.
```suggestion
        Book book1 = new Book();
        Book book2 = new Book();
        List<Book> books = List.of(book1, book2);
                Pageable pageable = PageRequest.of(0, 10);
        Page<Book> bookPage = new PageImpl<>(books, pageable, books.size());
        BookDto bookDto1 = new BookDto();
        BookDto bookDto2 = new BookDto();
        List<BookDto> expectedBooks = List.of(bookDto1, bookDto2);
        
        when(bookRepository.findAll(pageable)).thenReturn(bookPage);
        when(bookMapper.toDto(book1)).thenReturn(bookDto1);
        when(bookMapper.toDto(book2)).thenReturn(bookDto2);
        
        List<BookDto> result = bookService.findAll(pageable);
        assertEquals(expectedBooks, result);
        verify(bookRepository, times(1)).findAll(pageable);
        verify(bookMapper, times(1)).toDto(book1);
        verify(bookMapper, times(1)).toDto(book2);
```